qiime2/sdk/plugin_manager.py

```diff
@@ -105,25 +104,49 @@ def _integrate_plugin(self, plugin):
     # something wrong, the entire plugin fails to load any piece, like a
     # databases rollback/commit
 
-    @property
-    def importable_formats(self):
-        """Return formats that are importable.
+    def get_formats(self, *, include_all=False, plugin=None):
+        # Initialize formats dictionary
+        formats = {}
 
-        A format is importable in a QIIME 2 deployment if it can be transformed
-        into at least one of the canonical semantic type formats.
+        # Ternary operator to get all the plugins if plugin is None
+        plugins = [plugin] if plugin else self.plugins
 
-        """
-        importable_formats = {}
-        for name, record in self.formats.items():
-            from_type = transform.ModelType.from_view_type(
-                record.format)
-            for type_format in self.type_formats:
-                to_type = transform.ModelType.from_view_type(
+        # Get all formats if include_all is True
+        if include_all is True:
+            for plugin in plugins.values():
+                formats.update(plugin.formats)
+            return formats
+
+        # Loop through the plugins that we want to get the formats for those
+        # plugins
+        for plugin in plugins:
```

Comment:
I think the logic here can be simplified a bit --- this method should return all the formats that are implicated in a transformation. Let's chat about this IRL when you get a chance.